group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/TopicMetadata.java

```diff
@@ -40,23 +44,31 @@ public class TopicMetadata {
      */
     private final int numPartitions;
```

Comment:
Should we move this one as it can get it from `partitionRackInfo.size()`?

Reply:
yep removed numPartitions attribute but kept the numPartitions method to return the size of the map

Reply:
reverted this since we decided to keep numPartitions and pass an empty map to save space when no rack info is present